### PR TITLE
Trigger signing modals from approval requests

### PR DESCRIPTION
**Pull request by matthewwalsh0**
**Description**

Trigger the signature request modals using the generic `ApprovalController` requests rather than events specific to the `SignatureController`.

**Checklist**

* [ ] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/components/Nav/Main/RootRPCMethodsUI.js [hunk 1]
```diff
@@ -695,6 +709,27 @@ const RootRPCMethodsUI = (props) => {
...
+            origin: request.origin,
+          });
+          break;
+        case ApprovalTypes.ETH_SIGN_TYPED_DATA:
+          setSignMessageParams(requestData);
+          showPendingApprovalModal({
+            type: ApprovalTypes.ETH_SIGN_TYPED_DATA,
+            origin: request.origin,
+          });
+          break;
```
Comment (bergarces) [inline app/components/Nav/Main/RootRPCMethodsUI.js:732]:
```suggestion
        case ApprovalTypes.ETH_SIGN:
        case ApprovalTypes.PERSONAL_SIGN:
        case ApprovalTypes.ETH_SIGN_TYPED_DATA:
          setSignMessageParams(requestData);
          showPendingApprovalModal({
            type: request.type,
            origin: request.origin,
          });
          break;
```

Comment (matthewwalsh0) [reply app/components/Nav/Main/RootRPCMethodsUI.js]: Done.

**Review by bergarces — DISMISSED** [review]

File: ios/Podfile.lock [hunk 2]
```diff
@@ -874,4 +874,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 74e503a275b4ed097cb01c171027882e9cdc699d
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.12.1
```
Comment (OGPoyraz) [inline ios/Podfile.lock:877]: I guess this was not intentional

Comment (matthewwalsh0) [reply ios/Podfile.lock:877]: A previous commit bumped it so I updated my gem to match, but then another commit set it back to theirs so not sure which was intentional 😄 

Will remove mine and leave as is.

**Review by vinistevam — APPROVED** [review]

**Pull request merged**